File: leetcode/0507-perfect-number/Solution.java
class Solution {
    public boolean checkPerfectNumber(int num) {
        if (num <= 1) {
            return false;
        }

        long divisorSum = 1;
        for (long i = 2; i * i <= num; i++) {
            if (num % i == 0) {
                divisorSum += i;
                long other = num / i;
                if (other != i) {
                    divisorSum += other;
                }
            }
        }

        return divisorSum == num;
    }
}

File: leetcode/0507-perfect-number/solution.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num <= 1:
            return False

        divisor_sum = 1  # 1 always divides num (and we exclude num itself)
        i = 2
        while i * i <= num:
            if num % i == 0:
                divisor_sum += i
                other = num // i
                if other != i:
                    divisor_sum += other
            i += 1

        return divisor_sum == num
